### ISSUE TITLE
subspan axes and regridding: result smears out

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#975]
**Reported by ansley b manke on 5 Oct 2009 18:07 UTC**
(This may be related to #513)

Here is an example in the Users Guide demonstrating the @MAX regridding transformation, with some additions showing the results for a couple of weighted transforms @SUM and @NGD.  These show that the data is being spread out beyond the edges of the 1-point axis of the source grid.  The behavior is not particular to a 1-point axis; it occurs with other subspan modulo axes.

```
yes? define axis/x=163e/npoints=1 x1pt
yes? show axis x1pt
 name       axis              # pts   start                end
 X1PT      LONGITUDE            1mr   163E                 163E
   Axis span (to cell edges) = 1 (modulo length = 360)
yes? let var_1pt = randu(x[! a random value at a single coordinate
yes? list var_1pt
             VARIABLE : RANDU(X[GX=X1PT](gx=x1pt])))
             LONGITUDE: 163E
          0.4914

yes? define axis/x=161e:165e:1 x5pt
yes? show axis x5pt
 name       axis              # pts   start                end
 X5PT      LONGITUDE            5mr   161E                 165E
   Axis span (to cell edges) = 5 (modulo length = 360)

   ! wrong value with @MAX regridding
yes? list var_1pt[
             VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@max]))
                        regrid: 1 deg on X@MAX
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:     ....
 162E   / 2:     ....
 163E   / 3:  0.04804
 164E   / 4:     ....
 165E   / 5:     ....

   ! A weighted transform shows source data smearing out:
yes? list var_1pt[            VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@sum]
))
                        regrid: 1 deg on X@SUM
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:  0.00189
 162E   / 2:  0.00189
 163E   / 3:  0.04804
 164E   / 4:     ....
 165E   / 5:     ....

yes? list var_1pt[            VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@ngd]
)) (# of points)
                        regrid: 1 deg on X@NGD
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:  0.003
 162E   / 2:  0.003
 163E   / 3:  1.000
 164E   / 4:  0.000
 165E   / 5:  0.000
```

If the modulo attribute of the source axis is CANCELLED, the script works as it originally did.  Not necessary to cancel the modulo nature of the destination axis.

```

yes? define axis/x=163e/npoints=1 x1pt
yes? CANCEL AXIS/MODULO x1pt
yes? let var_1pt = randu(x[! a random value at a single coordinate
yes? list var_1pt
             VARIABLE : RANDU(X[GX=X1PT](gx=x1pt])))
             LONGITUDE: 163E
          0.4914

yes? define axis/x=161e:165e:1 x5pt

yes? list var_1pt[! same value embedded within 5 point axis
             VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@max]))
                        regrid: 1 deg on X@MAX
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:    ....
 162E   / 2:    ....
 163E   / 3:  0.4914
 164E   / 4:    ....
 165E   / 5:    ....

yes? list var_1pt[            VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@sum]
))
                        regrid: 1 deg on X@SUM
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:    ....
 162E   / 2:    ....
 163E   / 3:  0.4914
 164E   / 4:    ....
 165E   / 5:    ....

yes? list var_1pt[            VARIABLE : RANDU(X[GX=X1PT](gx=x5pt@ngd]
)) (# of points)
                        regrid: 1 deg on X@NGD
             SUBSET   : 5 points (LONGITUDE)
 161E   / 1:   ....
 162E   / 2:   ....
 163E   / 3:  1.000
 164E   / 4:   ....
 165E   / 5:   ....

```
 

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/1703

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 14 Oct 2009 17:20 UTC**
Here is a further example of this or something similar. A SET REGION that extends beyond the defined length of the destination subspan modulo axis causes the modulo void point to be saved as part of the result axis.

Look at the X axis coordinates:


```
yes? set region/x=5:26 

yes? define axis/units=degrees/x=-20:30/npoints=51 xsrc
yes? let median = x[define axis/units=degrees/x=7.25:25.25/npoints=37 xdst
yes? let evap = x[gx=xdst](gx=xsrc]

yes?)

yes? let p0 = median[save/clobber/file=es.nc p0
 LISTing to file es.nc

yes? sp ncdump es.nc 
netcdf es {
dimensions:
        XDST0_38 = 39 ;
variables:
        double XDST0_38(XDST0_38) ;
                XDST0_38:units = "DEGREES" ;
                XDST0_38:point_spacing = "even" ;
                XDST0_38:axis = "X" ;
                XDST0_38:modulo = 360. ;
        float P0(XDST0_38) ;
                P0:missing_value = -1.e+34f ;
                P0:_FillValue = -1.e+34f ;
                P0:long_name = "MEDIAN[GX=EVAP](gx=evap]
yes?)" ;

// global attributes:
                :history = "FERRET V6.3   14-Oct-09" ;
                :Conventions = "CF-1.0" ;
data:

 XDST0_38 = -163.75, 7.25, 7.75, 8.25, 8.75, 9.25, 9.75, 10.25, 10.75, 11.25, 
    11.75, 12.25, 12.75, 13.25, 13.75, 14.25, 14.75, 15.25, 15.75, 16.25, 
    16.75, 17.25, 17.75, 18.25, 18.75, 19.25, 19.75, 20.25, 20.75, 21.25, 
    21.75, 22.25, 22.75, 23.25, 23.75, 24.25, 24.75, 25.25, 196.25 ;

 P0 = -163.75, 7.25, 7.75, 8.25, 8.75, 9.25, 9.75, 10.25, 10.75, 11.25, 
    11.75, 12.25, 12.75, 13.25, 13.75, 14.25, 14.75, 15.25, 15.75, 16.25, 
    16.75, 17.25, 17.75, 18.25, 18.75, 19.25, 19.75, 20.25, 20.75, 21.25, 
    21.75, 22.25, 22.75, 23.25, 23.75, 24.25, 24.75, 25.25, _ ;
}

```


The same result occurs if we CANCEL REGION, but define the regridding as


```
let p0 = median[also if it's done as a qualifier on the SAVE,

```
yes? let p0 = median[gx=evap](gx=evap,x=5:26]
```

and)
yes? save/clobber/file=es.nc/x=5:26 p0
```

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 5 May 2016 17:53 UTC**
The first part of this is fixed along with the fixes to #2364, 2366.  Will open a new ticket with the SAVE issue.
